Add Trafikverket WeatherStation documentation

Qxlkdr opened the pull request:
**Description:**
Add Trafikverket WeatherStation documentation

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#12115

## Checklist:

- [X] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [X] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

Qxlkdr committed:
Create sensor.trafikverket_weatherstation.markdown.markdown

Qxlkdr committed:
Add trafikverket.png

trafikverket.png used for trafikverket_weatherstation sensor platform.

Qxlkdr committed:
api_key

Updated from api to api_key due to CONF_API_KEY.

Qxlkdr committed:
Removed text

Removed text which I used on GitHub for the custom component. Now needed to have for the official release.

Qxlkdr committed:
Example picture

Added an example picture to better understand how to get the weather station.

Qxlkdr committed:
Added example picture

Added an example picture to better understand how to get the weather station. What the user needs to copy and add to the YAML file.

Qxlkdr committed:
Added usage information

Added examples of the usage (why needing the data).

Qxlkdr committed:
linkable_title

Changed from normal titles to linkable_title in order to follow the documentation standards.

rytilahti commented on source/_components/sensor.trafikverket_weatherstation.markdown:
```diff
@@ -0,0 +1,66 @@
+- **scan_interval** (*Optional*): How frequently to query for new data (in seconds). Defaults to 300 seconds (5 minutes)
+
+##### {% linkable_title Get API key: %}
+[https://api.trafikinfo.trafikverket.se/](https://api.trafikinfo.trafikverket.se/)
```
Use full sentences when writing documentation.

Qxlkdr replied:
What do you mean? Please explain more in detail.

rytilahti replied:
Title could be "Obtaining API key" and description something like "Please visit <site> and register to obtain the API key" (maybe even explain the procedure & the limitations, if any, a bit?).

Qxlkdr replied:
Why want to have the description (whole sentence) as a link? Will check the procedure, if I need to mention anything. Don't think so but I will check.

Qxlkdr replied:
Whole sentence for all links (not http.......)? Just checking, I will change the map link as well in that case.

rytilahti replied:
It does not have to be as a link, but documentation is usually written as prose instead of just having links and listings.

Qxlkdr replied:
Updated, please review.

rytilahti commented on source/_components/sensor.trafikverket_weatherstation.markdown:
```diff
@@ -0,0 +1,66 @@
+- **api_key** (*Required*): API key from Trafikverket
+- **station** (*Required*): Name of the weather station
+- **type** (*Required*): Defines which temperature you want (`air` or `road`)
+- **scan_interval** (*Optional*): How frequently to query for new data (in seconds). Defaults to 300 seconds (5 minutes)
```
Configuration needs to be defined in a pre-defined manner, see https://home-assistant.io/developers/documentation/standards/#component-and-platform-pages and https://github.com/home-assistant/home-assistant.github.io/blob/next/source/_components/sensor.temper.markdown for an example.

Qxlkdr replied:
Why is this needed? Some components doesn't have it, like the Roth Touchline. Is it different requirements? I can change it, just curious to learn.

rytilahti replied:
I think it is to keep the documentation parseable (to allow consistent UI representations among components I suppose, in case the current one gets changed, it looks already different than those "old-style" ones.), but I don't know for sure either. I just saw that a while ago and updated my own PR accordingly.

rytilahti replied:
Another use-case I can think of is notifying the user / developer that the configuration format has changed, if it differs from the source code. This could be really helpful to avoid stale documentation, and should probably also be done on service calls (not relevant for this PR though).

Qxlkdr replied:
Updated to use `{% configuration %}`, please review.

rytilahti commented on source/_components/sensor.trafikverket_weatherstation.markdown:
```diff
@@ -0,0 +1,66 @@
+ha_release: 0.63.0
+---
+
+Showing weather information for air and road temperature provided by Trafikverket in Sweden. 
```
Trafikverket could be a link to their homepage here.

Qxlkdr replied:
Updated, please review.

rytilahti commented on source/_components/sensor.trafikverket_weatherstation.markdown:
```diff
@@ -0,0 +1,66 @@
+
+Showing weather information for air and road temperature provided by Trafikverket in Sweden. 
+
+Usage: Get weather data if you live near a weather station, just want to have the current temperature, needing the current temperature to know when your car heating system should start or when to leave for work due to queues (because the road is frozen).
```
This needs rephrasing, currently it's very unclear languagewise. Maybe have a list `Potential use-cases`?

Qxlkdr replied:
Updated, please review.

rytilahti commented on source/_components/sensor.trafikverket_weatherstation.markdown:
```diff
@@ -0,0 +1,66 @@
+<p class='img'>
+  <img src='{{site_root}}/images/screenshots/get_trafikverket_weather_station_example.png' />
+</p>
+
```
Same as above.

Qxlkdr replied:
Same as above, which one? Full sentence?

rytilahti replied:
Yes, using full sentences.

Qxlkdr replied:
Updated, please review.

rytilahti commented on source/_components/sensor.trafikverket_weatherstation.markdown:
````diff
@@ -0,0 +1,66 @@
+    station: Lanna
+    type: air
+    scan_interval: 600
+```
````
Is this necessary when an example configuration is already shown above?

Qxlkdr replied:
I can remove it if you insist but I think it's a good idea for the end user to see how to add multiple sensors for both `air` and `road`, but changing the scan interval at the same time.

Qxlkdr committed:
Updated based on feedback

Done
- Rewrote "Usage" into "Potential use cases"
- Linked to Trafikverket homepage
- Description instead of static links

Qxlkdr committed:
Configuration variables

Updated to use "{% configuration %}" instead of "Configuration variables" as plain text.

Qxlkdr commented:
@rytilahti I have updated all as requested. The only one I left was the examples, please advise. I can remove it but it doesn't hurt to give the end users a few more examples.

Qxlkdr committed:
HA release 0.64.0

Updated ha_release to 0.64.0 instead.

rytilahti commented:
I think extra examples do no harm, so I think this is fine now as it is.

Qxlkdr committed:
HA release 0.65.0

Updated ha_release to 0.65.0 since it wasn't merged in 0.64.0 release.

frenck committed:
:pencil2: Tiny improvements

frenck commented on source/_components/sensor.trafikverket_weatherstation.markdown:
````diff
@@ -0,0 +1,88 @@
+    scan_interval: 600
+```
+
+*A good practice is to use `secrets.yaml` for all your API keys, username and password. You can read more about secrets [here](https://home-assistant.io/docs/configuration/secrets/).*
````
Please remove this note, since it has nothing to do with this component.

Qxlkdr replied:
Thanks for reviewing my documentation. I have removed the note and updated `ha_release`, please have a look. Thanks in advance.

Qxlkdr replied:
**deploy/netlify is failing:**
*failed during stage 'deploying site': Invalid filename 'components/mqtt/#testing-your-setup.html'. Deployed filenames cannot contain # or ? characters*

I assume this is not relevant to my documentation?

frenck reviewed (changes requested):
Thank you @Qxlkdr, I've reviewed and left you one RFC.
Could you please take a look? 👍

Qxlkdr committed:
ha_release and note

Updated ha_release to 0.66.0 and removed a note about using secrets.yaml for all your API keys, usernames and passwords.

MartinHjelmare commented:
Please rebase your branch on latest `next` to let the build pass.

frenck commented:
Rebased the PR onto `next`.

frenck commented:
Thanks for this great first contribution to the documentation @Qxlkdr! 🏅  (and thanks for the component of course 😉 )

Going to merge this one in, since the parent PR has been merged already.

Qxlkdr commented:
@frenck 
Woho, thanks a lot. 
Will see what else I can build, hopefully for all users, not only us here in Sweden 😄
